umbraco/Umbraco-CMS
# Fix exceptions in Slider and Tags property value converters

### Prerequisites


If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/7622.

This includes the fixes done in the following PRs that targeted v8 (and were not merged in time and therefore got closed):
- https://github.com/umbraco/Umbraco-CMS/pull/8906
- https://github.com/umbraco/Umbraco-CMS/pull/8907

### Description
The property value converter (PVC) for the Slider data type would throw an `IndexOutOfRangeException` for existing stored single values after changing the configuration to return a range. With this PR in place, it will now return a range value (as expected per configuration) with the single value as both minimum and maximum.

The PVC for the Tags data type would throw a `NullReferenceException` when values are stored as JSON and the existing stored value was an empty string, as it can't deserialize an empty string to a string array.

I've also cleaned up the data type lookup and configuration caching, as we have direct access to this since v8.

To test the slider:
1. Add a property with a Slider data type to a document and save/publish a single value (e.g.`5`) ;
2. Render this value in the template using e.g. `@Model.Value("slider")`: this should display the value;
3. Enable range on the Slider data type and reload the rendered template:
   - Without this PR it will throw an `IndexOutOfRangeException`;
   - With this PR it will return a range value with both minimum and maximum values set to the single value (rendered as `5,5`).

Testing the Tags PVC is a bit harder, since PR https://github.com/umbraco/Umbraco-CMS/pull/11806 ensured empty values aren't stored anymore (which also makes the issue less likely to occur). Because the code changes are very minor, a code review should probably suffice.
